Modules/Loadable/SubjectHierarchy/Testing/Python/SubjectHierarchyGenericSelfTest.py

```diff
@@ -428,9 +410,135 @@ def section_TestCircularParenthood(self):
     sceneItemID = shNode.GetSceneItemID()
     mainfolder_ID = shNode.CreateFolderItem(sceneItemID, "Main Folder")
     subfolder_ID = shNode.CreateFolderItem(sceneItemID, "Sub Folder")
-    shNode.SetItemParent(subfolder_ID, mainfolder_ID) # Regular hiearchy setting
+    shNode.SetItemParent(subfolder_ID, mainfolder_ID) # Regular hierarchy setting
     shNode.SetItemParent(mainfolder_ID, subfolder_ID) # Makes slicer crash instead of returning an error
 
+  # ------------------------------------------------------------------------------
+  def section_AttributeFilters(self):
+    self.delayDisplay("Attribute filters",self.delayMs)
+
+    import SampleData
+    sceneFile = SampleData.downloadFromURL(
+      fileNames=self.attributeFilterTestSceneFileName,
+      uris=self.attributeFilterTestSceneFileUrl,
+      # loadFiles=True,
+      checksums=self.attributeFilterTestSceneChecksum)[0]
+    if not os.path.exists(sceneFile):
+      logging.error('Failed to download attribute filter test scene to path ' + str(sceneFile))
+    self.assertTrue(os.path.exists(sceneFile))
+
+    slicer.mrmlScene.Clear()
+    ioManager = slicer.app.ioManager()
+    ioManager.loadFile(sceneFile)
+
+    shNode = slicer.mrmlScene.GetSubjectHierarchyNode()
+
+    # Check scene validity
+    self.assertEqual(114, slicer.mrmlScene.GetNumberOfNodes())
+    self.assertEqual(9, shNode.GetNumberOfItems())
+    self.assertEqual(slicer.mrmlScene.GetNumberOfNodesByClass('vtkMRMLMarkupsNode'), 7)
+
+    # Create test SH tree view
+    shTreeView = slicer.qMRMLSubjectHierarchyTreeView()
+    shTreeView.setMRMLScene(slicer.mrmlScene)
+    shTreeView.show()
+
+    shProxyModel = shTreeView.sortFilterProxyModel()
+
+    def testAttributeFilters(filteredObject, proxyModel):
+      # Check include node attribute name filter
```

Comment:
It could be nice if you could add here in a comment attributes of those 9 nodes so that the test can be verified by code review.